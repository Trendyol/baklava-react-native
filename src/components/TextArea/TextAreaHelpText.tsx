import React from 'react';
import Box from '../Box/Box';
import Text from '../Text/Text';
import { getHelpText, getTextColor } from './utils';

export const TextAreaHelpText = React.memo(
  ({
    helpText,
    errorState,
    errorText,
  }: {
    helpText?: string | null;
    errorState: boolean;
    errorText?: string | null;
  }): JSX.Element | null => {
    if (!helpText && !errorText) {
      return null;
    }

    const textColor = getTextColor({ errorState });

    const content = getHelpText({
      helpText,
      errorText,
      errorState,
    });

    return content ? (
      <Box mt="3xs">
        <Text variant="subtitle04Regular" color={textColor}>
          {content}
        </Text>
      </Box>
    ) : null;
  },
);
