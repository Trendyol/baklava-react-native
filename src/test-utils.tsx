import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
