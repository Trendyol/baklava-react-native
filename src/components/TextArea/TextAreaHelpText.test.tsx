import React from 'react';
import { render } from '../../test-utils';
import { TextAreaHelpText } from './TextAreaHelpText';

describe('TextArea Help Text', () => {
  let helpText: string | null;
  let errorText: string | null;
  let errorState: boolean;

  test('should render help text correctly', () => {
    // given
    helpText = 'help text';
    errorText = 'error text';
    errorState = false;

    // when
    const { toJSON } = render(
      <TextAreaHelpText
        helpText={helpText}
        errorState={errorState}
        errorText={errorText}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render null when not has help text and error text', () => {
    // given
    helpText = null;
    errorText = null;
    errorState = false;

    // when
    const { toJSON } = render(
      <TextAreaHelpText
        helpText={helpText}
        errorState={errorState}
        errorText={errorText}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render null when not has help text', () => {
    // given
    helpText = '';
    errorText = 'error text';
    errorState = false;

    // when
    const { toJSON } = render(
      <TextAreaHelpText
        helpText={helpText}
        errorState={errorState}
        errorText={errorText}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
