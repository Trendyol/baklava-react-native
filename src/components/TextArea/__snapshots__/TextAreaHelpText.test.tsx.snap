// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Help Text should render help text correctly 1`] = `
<View
  accessible={false}
  style={
    Array [
      Object {
        "marginTop": 4,
      },
    ]
  }
>
  <Text
    accessible={false}
    style={
      Array [
        Object {
          "color": "#95A1B5",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 14,
        },
      ]
    }
  >
    help text
  </Text>
</View>
`;

exports[`TextArea Help Text should render null when not has help text 1`] = `null`;

exports[`TextArea Help Text should render null when not has help text and error text 1`] = `null`;
